File: src/ProjectR.Scripting/content/scripts/spells/lua/MagicMissile.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Magic Missile"
end

function GetDescription()
    return "An arcane missile. Not really good for much, but cheap."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .16
end

function GetDelay()
  return .5
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { ARC }
end

function SpellEffect()
  local damage = (3.6 * aMD * (aARC/100) - dMR) * (100/dARC)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/MagicMissile.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class MagicMissile : SpellScriptBase
    {
        public override string Name { get { return "Magic Missile"; } }
        public override string Description { get { return "An arcane missile. Not really good for much, but cheap."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.ARC, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 16; } }
        public override double Delay { get { return .5; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (3.6 * cMD * (cARC / 100) - tMR) * (100 / tARC);
        }
    }
}